zerver/lib/create_user.py

```diff
@@ -81,12 +81,17 @@ def create_user(email: str, password: Optional[str], realm: Realm,
                 bot_type: Optional[int] = None,
                 bot_owner: Optional[UserProfile] = None,
                 tos_version: Optional[str] = None, timezone: str = "",
+                user_avatar_url: str='',
                 avatar_source: str = UserProfile.AVATAR_FROM_GRAVATAR,
                 is_mirror_dummy: bool = False,
                 default_sending_stream: Optional[Stream] = None,
                 default_events_register_stream: Optional[Stream] = None,
                 default_all_public_streams: Optional[bool] = None,
                 source_profile: Optional[UserProfile] = None) -> UserProfile:
+    # If a user is being created after authenticating using GitHub, than it doesn't make
```

Comment:
Should have noticed earlier, but I don't think we need the detail that the user got here through github in this function. Especially since this comment will become wrong the moment a different codepath calls this function with `user_avatar_url`. It should be sufficient to comment that it doesn't make sense for the caller to give an avatar_url to use and at the same request gravatar as the avatar source :thinking: